main.js

```diff
@@ -1304,6 +1294,17 @@ installSettingsSetter('incoming-call-notification');
 
 // These ones are different because its single source of truth is userConfig,
 // not IndexedDB
+ipc.on('get-stay-in-tray', event => {
+  event.sender.send(
+    'get-success-stay-in-tray',
+    null,
+    userConfig.get('stayInTray') || false
```

Comment:
I would recommend setting  the Stay In Tray option default to True as it is most likely the expected behaviour users expects

Reply:
Please not. There are desktops who do not support a tray. Maybe check for tray support and do not enable it in case there is none or add a feature blocklist for some desktop environments, like GNOME-Shell.

Reply:
@fabiscafe thanks for your input.

Do you know how electron behave in OS that doesn't support tray ?

Reply:
The Gnome-Shell doesn't have a native tray, but there's an extension that can be installed that fixes that issue. I'm using Gnome-Shell with the extension, and I also vote to make Stay In Tray the default option. It's not your fault that Gnome doesn't want to have a tray, while most desktops presumably have one. And also, because Signal IMHO needs a larger user base, and having good default options like this hopefully helps.

> Do you know how electron behave in OS that doesn't support tray?

It still runs in the background, but you simply won't see a tray icon. That's all.

Reply:
> there's an extension that can be installed

That wont work on Wayland.

> that fixes that issue

Its intentional, not an issue.

Reply:
> @fabiscafe thanks for your input.
> 
> Do you know how electron behave in OS that doesn't support tray ?

For electron itself I have no idea. Just a few apps

Discord: checks for tray support and disabled minimize to tray if not found
rocket.chat and Slack: running in the background regardless

Hm I think thats all I know about electron based apps with tray icons…

So the ideal solution would be to check for support and disable minimize to tray completely if not there.

In any chance, please
* dont implement functionality that would require a tray
* name the feature "stay in background", or something that doesn't refers to a system tray
* if you let the app run in the background, make sure that it is not possible to launch a secound session of the app when someone launches the app again from the systemmenu. (_I think thats already the case for Signal_)


What just came to my mind:
Instead of enable or disable by default: ask the user what should be set on first click on close. :+1: